src/harness/testcases/WINNF_FT_S_HBT_testcase.py

```diff
@@ -0,0 +1,155 @@
+#    Copyright 2016 SAS Project Authors. All Rights Reserved.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+import json
+import os
+import sas
+from  util import winnforum_testcase, makePpaAndPalRecordsConsistent
+import sas_testcase
+from datetime import datetime, timedelta
+import time
+
+
+class HeartbeatTestcase(sas_testcase.SasTestCase):
+  def setUp(self):
+    self._sas, self._sas_admin = sas.GetTestingSas()
+    self._sas_admin.Reset()
+
+  def tearDown(self):
+    pass
+
+  def getHeartBeatRequest(self,cbsd_ids, grant_ids, previousTransmitExpireTime):     
```

Comment:
Do we really need all of this machinery? Since it's a very simple message, we usually just create the heartbeat request in the test case itself so that it's easier to read (and verify that it's doing the right thing). This is especially true for the HBT test cases where we often want to create the request in a very specific way. Is there a reason to do something else?

I'm also worried about the correctness of this function, but I'll hold off on those comments until we establish that it's a useful function to have.

(Same comment for the following function.)

Reply:
you are right, for the next function, because it is used only once in the test, but this function is used two times in the test, and i think it will be useful for other HB test cases so don't you think it is better to reuse it?
I think for this function it is also related to the new added machineryfor choosing the right status of the grant inside the HB request.
we need to verify that it is synchronized with the status in SAS" especially after the IAP which can take some time", other wise the SAS will send response 502, right?
so if you don't agree on this status machinery, we don't need to have this function.

Reply:
Aha, now I understand your concern. I think the machinery is unnecessary during the first heartbeat request (we know that it should always be GRANTED). For the second heartbeat request, I think we can safely use GRANTED as well -- even if the SAS thinks the Grant is AUTHORIZED, it should not give an error. What do you think?

Reply:
yeah, you are right, I thought the CBSD should always send in HB the status which exists in the SAS otherwise the SAS will send error(502), but I think you are right if the CBSD sends granted status in the HB request and its status on the SAS is Authorized it is safe, and there should be no error.
We don't need this machinery then, thank you.